public/apps/configuration/utils/request-utils.ts

```diff
@@ -13,29 +13,95 @@
  *   permissions and limitations under the License.
  */
 
-import { HttpStart, HttpHandler } from 'opensearch-dashboards/public';
+import { HttpStart, HttpHandler, HttpFetchQuery } from 'opensearch-dashboards/public';
 
-export async function request<T>(requestFunc: HttpHandler, url: string, body?: object): Promise<T> {
-  if (body) {
-    return (await requestFunc(url, { body: JSON.stringify(body) })) as T;
-  }
-  return (await requestFunc(url)) as T;
+interface BaseRequestParams {
+  url: string;
+  body?: object;
 }
 
-export async function httpGet<T>(http: HttpStart, url: string): Promise<T> {
```

Comment:
ok, I see why the RequestType interface was around in a previous commit, I just thought the abstraction was confusing.

Its possible in typescript to define types inline within a fn declaration (not saying that is optimal here, just that its possible.

```
export async function httpGet<T>({http, url, body}: {http: HttpStart, url: string, body?: object}): Promise<T> {
    ...
}
```

Reply:
Yes I agree - I think overall we may not even need these helper functions, as all they are doing is for example extracting the `get` method off of the httpStart object. However, I didn't want to change the code too much and decided to keep things mostly the same, but make things more clear since we are introducing more variables. The way the code currently is, I think it benefits from a type hierarchy. For example, all types will have a url and optional body, then we have query that is being constructed from the RequestContext, then we have optional ignores if we need to make the call safe against some error codes, etc.... I think there is a good amount of unused code/utility functions that only are used in one place. I will create an issue to clean those up later on.

Reply:
Created issue here: https://github.com/opensearch-project/security-dashboards-plugin/issues/1906